astropy/coordinates/representation/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""
+In this module, we define the coordinate representation classes, which are
+used to represent low-level cartesian, spherical, cylindrical, and other
+coordinates.
+"""
+from .base import BaseRepresentationOrDifferential, BaseRepresentation, BaseDifferential
+from .cartesian import CartesianRepresentation, CartesianDifferential
+from .cylindrical import CylindricalRepresentation, CylindricalDifferential
+from .spherical import (
+    SphericalRepresentation,
+    UnitSphericalRepresentation,
+    RadialRepresentation,
+    PhysicsSphericalRepresentation,
+    BaseSphericalDifferential,
+    BaseSphericalCosLatDifferential,
+    SphericalDifferential,
+    SphericalCosLatDifferential,
+    UnitSphericalDifferential,
+    UnitSphericalCosLatDifferential,
+    RadialDifferential,
+    PhysicsSphericalDifferential,
+)
+
+# The following imports are included for backwards compatibility.
+# isort: split
+from .base import DIFFERENTIAL_CLASSES
+from .base import DUPLICATE_REPRESENTATIONS
+from .base import REPRESENTATION_CLASSES
```

Comment:
Would it be worthwhile to make these read-only?

```suggestion
from types import MappingProxyType

from .base import MappingProxyType(DIFFERENTIAL_CLASSES)
from .base import MappingProxyType(DUPLICATE_REPRESENTATIONS)
from .base import MappingProxyType(REPRESENTATION_CLASSES)
```

Or to deprecate their access from here?

Reply:
I agree it should be deprecated, which is relatively easy to do with `__getattr__`, but think it is best to do as a separate PR after this, so one would not be distracted by all the other changes.